src/main/java/br/com/sunty/dao/CategoryDao.java

```diff
@@ -0,0 +1,42 @@
+package br.com.sunty.dao;
+
+import br.com.sunty.models.category.Category;
+import br.com.sunty.models.category.SubCategory;
+import br.com.sunty.models.course.Course;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+
+public class CategoryDao {
+
+    private EntityManager entityManager;
+
+    public CategoryDao(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public void create(Category category) {
+        this.entityManager.persist(category);
+    }
+
+    public void update(Category category) {
+        this.entityManager.merge(category);
+    }
+
+    public void delete(Category category) {
+        category = entityManager.merge(category);
+        this.entityManager.remove(category);
+    }
+
+    public Category findByUrlCode(String urlCode) {
+        String jpql = "SELECT c FROM Category c where c.urlCode=:urlCode";
+        return entityManager.createQuery(jpql, Category.class)
+                .setParameter("urlCode", urlCode)
+                .getSingleResult();
+    }
+
+    public List<Category> findAllActiveOrdered() {
```

Comment:
Somente sugestão:
```suggestion
    public List<Category> findAllByiIsActiveIsTrueOrderByOrderToShowAsc() {
```

Na Alura usamos o [Spring Data JPA](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation), então fazemos as queries dessa forma.